diff --git a/dist/module.d.mts b/dist/module.d.mts
index 4981b0e4e6bd39552de9571f256e01053fc8c6ff..2f2426ebc548c69ae3a60b79ea7e78e8255f6249 100644
--- a/dist/module.d.mts
+++ b/dist/module.d.mts
@@ -32,6 +32,12 @@ interface ModuleOptions {
         theme?: MdcThemeOptions;
         /**
          * Languages to be bundled loaded by Shiki
+         *
+         * All languages used has to be included in this list at build time, to create granular bundles.
+         *
+         * Unlike the `preload` option, when this option is provided, it will override the default languages.
+         *
+         * @default ['js','jsx','json','ts','tsx','vue','css','html','vue','bash','md','mdc','yaml']
          */
         langs?: (BundledLanguage | LanguageRegistration)[];
         /**
@@ -62,6 +68,7 @@ interface ModuleOptions {
     };
 }
 
+declare const DefaultHighlightLangs: BundledLanguage[];
 declare const _default: _nuxt_schema.NuxtModule<ModuleOptions>;
 
 declare module '@nuxt/schema' {
@@ -92,4 +99,4 @@ declare module '@nuxt/schema' {
     }
 }
 
-export { _default as default };
+export { DefaultHighlightLangs, _default as default };
diff --git a/dist/module.d.ts b/dist/module.d.ts
index 2173adeab0cf081fb085ff2b1d9a8f18af213bc6..67d498257c225de9c8043d3afd182d3d5d13ddaf 100644
--- a/dist/module.d.ts
+++ b/dist/module.d.ts
@@ -32,6 +32,12 @@ interface ModuleOptions {
         theme?: MdcThemeOptions;
         /**
          * Languages to be bundled loaded by Shiki
+         *
+         * All languages used has to be included in this list at build time, to create granular bundles.
+         *
+         * Unlike the `preload` option, when this option is provided, it will override the default languages.
+         *
+         * @default ['js','jsx','json','ts','tsx','vue','css','html','vue','bash','md','mdc','yaml']
          */
         langs?: (BundledLanguage | LanguageRegistration)[];
         /**
@@ -62,6 +68,7 @@ interface ModuleOptions {
     };
 }
 
+declare const DefaultHighlightLangs: BundledLanguage[];
 declare const _default: _nuxt_schema.NuxtModule<ModuleOptions>;
 
 declare module '@nuxt/schema' {
@@ -92,4 +99,4 @@ declare module '@nuxt/schema' {
     }
 }
 
-export { _default as default };
+export { DefaultHighlightLangs, _default as default };
diff --git a/dist/module.mjs b/dist/module.mjs
index a338b5bb3396404c0975bab62c54359b35bebfc9..92b7458931588130effec06042a70efb0364669b 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -159,6 +159,21 @@ function processUnistPlugins(plugins) {
   return { imports, definitions };
 }
 
+const DefaultHighlightLangs = [
+  "js",
+  "jsx",
+  "json",
+  "ts",
+  "tsx",
+  "vue",
+  "css",
+  "html",
+  "vue",
+  "bash",
+  "md",
+  "mdc",
+  "yaml"
+];
 const module = defineNuxtModule({
   meta: {
     name: "@nuxtjs/mdc",
@@ -354,19 +369,11 @@ function resolveOptions(options) {
       default: "github-light",
       dark: "github-dark"
     };
-    options.highlight.langs ||= [
-      "js",
-      "ts",
-      "vue",
-      "css",
-      "html",
-      "vue",
-      "shell"
-    ];
+    options.highlight.langs ||= DefaultHighlightLangs;
     if (options.highlight.preload) {
       options.highlight.langs.push(...options.highlight.preload || []);
     }
   }
 }
 
-export { module as default };
+export { DefaultHighlightLangs, module as default };
diff --git a/dist/types.d.mts b/dist/types.d.mts
index a6d26a8e28e231d7818734a911c50115836bc167..e96a65f0b52aad3a6c8e1b744caa971b5aa5993d 100644
--- a/dist/types.d.mts
+++ b/dist/types.d.mts
@@ -5,4 +5,4 @@ import type {  } from './module.js'
 
 
 
-export type { default } from './module.js'
+export type { DefaultHighlightLangs, default } from './module.js'
diff --git a/dist/types.d.ts b/dist/types.d.ts
index 793a5c9634bab6bee215b1c5a49926bd733163a7..a9e4d1746fbc90a669d3cee634a6dc4fa817fcb2 100644
--- a/dist/types.d.ts
+++ b/dist/types.d.ts
@@ -5,4 +5,4 @@ import type {  } from './module'
 
 
 
-export type { default } from './module'
+export type { DefaultHighlightLangs, default } from './module'
