diff --git a/dist/module.mjs b/dist/module.mjs
index 6b9128e76c3921301cdec76e267063274224d1fe..e9c6001adbc49033c5a8df96bf5b95751ef09eb9 100644
--- a/dist/module.mjs
+++ b/dist/module.mjs
@@ -30,7 +30,7 @@ const module = defineNuxtModule({
     nuxt.options.appConfig.ui = defu(nuxt.options.appConfig.ui || {}, getDefaultUiConfig(options.theme.colors));
     nuxt.options.app.rootAttrs = nuxt.options.app.rootAttrs || {};
     nuxt.options.app.rootAttrs.class = [nuxt.options.app.rootAttrs.class, "isolate"].filter(Boolean).join(" ");
-    nuxt.hook("vite:extendConfig", async (config) => {
+    nuxt.hook("vite:extend", async ({ config }) => {
       const plugin = await import('@tailwindcss/vite').then((r) => r.default);
       config.plugins ||= [];
       config.plugins.push(plugin());
