diff --git a/dist/presets/vercel/utils.mjs b/dist/presets/vercel/utils.mjs
index 2c8e5031e878465a4cb5943cff7d8fcd49e0f15f..a9c6bd69920d37ee7feac7b5903b2113d18a2467 100644
--- a/dist/presets/vercel/utils.mjs
+++ b/dist/presets/vercel/utils.mjs
@@ -1,10 +1,10 @@
-import fsp from "node:fs/promises";
-import { defu } from "defu";
-import { writeFile } from "nitropack/kit";
-import { dirname, relative, resolve } from "pathe";
-import { joinURL, withLeadingSlash, withoutLeadingSlash } from "ufo";
-import { isTest } from "std-env";
-import { createRouter as createRadixRouter, toRouteMatcher } from "radix3";
+import fsp from "node:fs/promises"
+import { defu } from "defu"
+import { writeFile } from "nitropack/kit"
+import { dirname, relative, resolve } from "pathe"
+import { joinURL, withLeadingSlash, withoutLeadingSlash } from "ufo"
+import { isTest } from "std-env"
+import { createRouter as createRadixRouter, toRouteMatcher } from "radix3"
 const SUPPORTED_NODE_VERSIONS = [18, 20, 22];
 const FALLBACK_ROUTE = "/__fallback";
 const ISR_SUFFIX = "-isr";
@@ -136,6 +136,21 @@ function generateBuildConfig(nitro, o11Routes) {
         }
         return route;
       }),
+      // skew protection
+      ...process.env.VERCEL_SKEW_PROTECTION_ENABLED ? [{
+        src: '/.*',
+        has: [
+          {
+            type: 'header',
+            key: 'Sec-Fetch-Dest',
+            value: 'document'
+          }
+			  ],
+        headers: {
+				  'Set-Cookie': `__vdpl=${process.env.VERCEL_DEPLOYMENT_ID}; Path=${nitro.options.baseURL}; SameSite=Strict; Secure; HttpOnly`
+        },
+        continue: true
+      }] : [],
       // Public asset rules
       ...nitro.options.publicAssets.filter((asset) => !asset.fallthrough).map((asset) => joinURL(nitro.options.baseURL, asset.baseURL || "/")).map((baseURL) => ({
         src: baseURL + "(.*)",
