diff --git a/dist/runtime/server/plugins/llms.js b/dist/runtime/server/plugins/llms.js
index 9f37e2a6e9095eaa980bd3186718939a04bab315..cc810fdcc06d55e8419b768dfd5418889c1cc1d9 100644
--- a/dist/runtime/server/plugins/llms.js
+++ b/dist/runtime/server/plugins/llms.js
@@ -163,7 +163,7 @@ export default defineNitroPlugin((nitroApp) => {
       }
     }
   }));
-  if (["nitro-prerender", "nitro-dev"].includes(import.meta.preset)) {
+  if (import.meta.preset === "nitro-prerender") {
     nitroApp.hooks.hook("beforeResponse", (event) => {
       if (event.path === "/" || event.path === "/llms.txt") {
         appendHeader(event, "x-nitro-prerender", Array.from(prerenderPaths));
